core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -55,6 +60,15 @@ class EventRepository {
 
     private static final String CHARSET_UTF_8 = "UTF-8";
 
+    // Handler instance associated with the main thread
+    static final Handler HANDLER =
+            new Handler(Looper.getMainLooper()) {
+                @Override
+                public void handleMessage(Message msg) {
+                    throw new AssertionError("Unknown handler message received: " + msg.what);
```

Comment:
This might lead to crash, and as far as I remember we decided not to crash the app under any circumstances.

Reply:
changed it log an error instead of throwing an error